src/wh_transport_mem.c

```diff
@@ -101,30 +106,43 @@ int wh_TransportMem_SendRequest(void* c, uint16_t len, const void* data)
         return WH_ERROR_BADARGS;
     }
 
-    /* Read current CSR's */
-    resp.u64 = context->resp->u64;
-    req.u64 = context->req->u64;
+    /* Read current CSR's. ctx_req does not need to be invalidated */
+    XMEMFENCE();
+    XCACHEINVLD(ctx_resp);
+    XCACHEINVLD(ctx_req);
+
+    resp.u64 = ctx_resp->u64;
+    req.u64 = ctx_req->u64;
 
     /* Has server completed with previous request */
     if (req.s.notify != resp.s.notify) {
         return WH_ERROR_NOTREADY;
     }
 
     if ((data != NULL) && (len != 0)) {
-        memcpy((void*)context->req_data, data, len);
+        wh_Utils_memcpy_flush((void*)context->req_data, data, len);
     }
+
+    /* Ensure the memcpy is complete */
+    XMEMFENCE();
```

Comment:
If we are always fencing after a memcpy, should we just pull this into `wh_Utils_memcpy_flush()`?

Reply:
This is a bug.  The fence needed to be after the memset/cpy but before the flush.  Updated.